packages/odyssey-react/src/components/TextInput/index.tsx

```diff
@@ -0,0 +1,159 @@
+/*!
+ * Copyright (c) 2021-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import React, { useMemo } from 'react';
+import type { FunctionComponent, ChangeEvent, ReactNode } from 'react';
+import { nanoid } from 'nanoid';
+const buildId = nanoid.bind(null, 6);
+
+export type TextInputProps = {
+  /**
+   * The underlying input element id attribute. Automatically generated if not provided.
+   */
+  id?: string,
+
+  /**
+   * The underlying input element type.
+   * @default text
+   */
+  type?: 'text' | 'email' | 'url' | 'tel' | 'search' | 'password',
+
+  /**
+   * The form field label.
+   */
+  label: string,
```

Comment:
Currently, this is associated with a separate Field or Field Label component. I'm 💯 on including it here to get the component up and running, as we'll be reworking the Form structure anyway. Just wanted to note the difference in structure in case you had thoughts on the future structure.

Reply:
@edburyenegren-okta are you saying there is a separate component that isn't exposed on the Odyssey doc site?

I'm +1 on using composition for labels here. This allows the consumer to control things like screen reader only labels easily without the need for a complicated series of if/else props that need to work together. Its a more powerful and flexible API for us to design.

Reply:
It's documented as Field Label currently (http://localhost:8080/components/field-labels/). 

@arnoldsandoval-okta ran into some structuring issues when trying to use the same component across Form Element use cases, which is why we have [an open item](https://oktainc.atlassian.net/browse/OKTA-398354) for revisiting Form structure.

Worth noting that a request from users so far is that things like this _do_ happen automatically, especially if there are a11y implications. Teams (thus far) do not want compliance-related behavior left to composition.

Reply:
Ah yes the Field Label thank you. What do you mean by happen automatically?